Makefile

```diff
@@ -1,34 +1,88 @@
-black:
-	black . --line-length 99
+SHELL = /bin/bash
 
-black-check:
-	black . --line-length 99 --check
+.SILENT:
 
-test:
-	coverage run --include 'lbz/*' -m pytest "tests"
-	coverage report --skip-covered
 
-build:
-	python setup.py sdist bdist_wheel
-
-upload:
-	python3 -m twine upload dist/*
+###############################################################################
+# Python Requirements
+# -----------------------------------------------------------------------------
+# TODO: Start using pip-tools as a dependency freezing tool
```

Comment:
_FYI:_ This pull request is quite big without `pip-tools` I didn't want to go too far with changes 😉